relay-general/src/protocol/thread.rs

```diff
@@ -69,6 +69,21 @@ impl Empty for ThreadId {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Empty, FromValue, IntoValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct LockReason {
+    #[metastructure(field = "type", required = "true")]
+    pub ty: Annotated<u64>,
+    #[metastructure(skip_serialization = "empty")]
+    pub address: Annotated<String>,
+    #[metastructure(skip_serialization = "empty")]
+    pub package_name: Annotated<String>,
+    #[metastructure(skip_serialization = "empty")]
+    pub class_name: Annotated<String>,
+    #[metastructure(skip_serialization = "empty")]
+    pub thread_id: Annotated<ThreadId>,
```

Comment:
This allows outdated external relays to forward new fields if we ever add them:

```suggestion
    pub thread_id: Annotated<ThreadId>,
    
    /// Additional arbitrary fields for forwards compatibility.
    #[metastructure(additional_properties)]
    pub other: Object<Value>,
```